aptos-move/framework/aptos-framework/sources/PropertyList.move

```diff
@@ -0,0 +1,182 @@
+/// This Data Structure is designed to hold properties.
+/// It only support
+///     1. adding new property.
+///     2. update the value of existing property
+///     3. get the value of a property
+module AptosFramework::PropertyList {
+    use Std::Vector;
+    use Std::ASCII::String;
+    use AptosFramework::SimpleMap::{Self, SimpleMap};
+
+    const MAX_PROPERTY_MAP_SIZE: u64 = 1000;
+    const EKEY_AREADY_EXIST_IN_PROPERTY_MAP: u64 = 1;
+    const EPROPERTY_NUMBER_EXCEED_LIMIT: u64 = 2;
+    const EPROPERTY_NOT_EXIST: u64 = 3;
+    const EKEY_COUNT_NOT_MATCH_VALUE_COUNT: u64 = 4;
+    const EKEY_COUNT_NOT_MATCH_TYPE_COUNT: u64 = 5;
+
+    struct PropertyList has store {
+        map: SimpleMap<String, PropertyValue>
+    }
+
+    struct PropertyValue has store, copy, drop {
+        value: vector<u8>,
+        type: String,
+    }
+
+    public fun new(
+        default_keys: &vector<String>,
+        default_values: &vector<vector<u8>>,
+        default_types: &vector<String>
```

Comment:
I don't think these are really defaults, so we can strike that from the api. They are the values at creation time. I think the default gets a little to close to how you think about it and might not be how other folks that look at the API look at it. Because another term is "global" or "original", so "" is probably okay.

Reply:
also I might recommend that we not pass these in by reference, I suspect more often than not, they will be from script functions and we'd be making unnecessary copies.

in general, I think it is best to only pass by reference if you won't need a copy or copy isn't available. sometimes.

Reply:
done